scene/resources/material.cpp

```diff
@@ -1139,7 +1139,7 @@ void BaseMaterial3D::_update_shader() {
 
 	if (!RenderingServer::get_singleton()->is_low_end() && features[FEATURE_HEIGHT_MAPPING] && !flags[FLAG_UV1_USE_TRIPLANAR]) { //heightmap not supported with triplanar
 		code += "	{\n";
-		code += "		vec3 view_dir = normalize(normalize(-VERTEX)*mat3(TANGENT*heightmap_flip.x,-BINORMAL*heightmap_flip.y,NORMAL));\n"; // binormal is negative due to mikktspace, flip 'unflips' it ;-)
+		code += "		vec3 view_dir = normalize(normalize(-VERTEX+EYE_OFFSET)*mat3(TANGENT*heightmap_flip.x,-BINORMAL*heightmap_flip.y,NORMAL));\n"; // binormal is negative due to mikktspace, flip 'unflips' it ;-)
```

Comment:
That's a pre-existing problem, but for a future refactoring pass, we try to format generated shader code the same way that our .glsl files are formatted. So this line needs a lot more spaces, and it will make it a lot more readable too :)

Reply:
Added some spaces, let me know if that was what you had in mind?

Reply:
Yeah that looks good.